androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -53,11 +58,55 @@ class AgendaDetailsViewModel(
         )
     val screenStateFlow = _screenStateFlow.asStateFlow()
 
-    private val _isDeleteSuccess = MutableStateFlow(false)
-    val isDeleteSuccess = _isDeleteSuccess.asStateFlow()
+    private val _isDeleteSuccessFlow = MutableStateFlow(false)
+    val isDeleteSuccessFlow: StateFlow<UiEvent<Boolean>> =
+        _isDeleteSuccessFlow
+            .map {
+                object : UiEvent<Boolean> {
+                    override val data: Boolean
+                        get() = it
+                    override val onConsume: () -> Unit
+                        get() = {
+                            _isDeleteSuccessFlow.update { false }
+                        }
+                }
+            }.stateIn(
+                viewModelScope,
+                SharingStarted.Lazily,
+                object : UiEvent<Boolean> {
+                    override val data: Boolean
+                        get() = false
+                    override val onConsume: () -> Unit
+                        get() = {
+                            _isDeleteSuccessFlow.update { false }
+                        }
+                },
+            )
```

Comment:
Don’t you think this is a bit verbose? You can use a simple channel and a flow for such one-time events. You wouldn’t then require a state to observe, but rather just a normal flow which you can collect inside your UI and react to different types of one-time events. To collect them safely, you can use the following composable.

```
@Composable
private fun <T> ObserveAsEvents(flow: Flow<T>, onEvent: (T) -> Unit) {
    val lifecycleOwner = LocalLifecycleOwner.current
    LaunchedEffect(flow, lifecycleOwner.lifecycle) {
        lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
            withContext(Dispatchers.Main.immediate) {
                flow.collect(onEvent)
            }
        }
    }
}
```

Your event model can then look like this:

```
sealed interface LoginEvent {
    data object OnLoginSuccess : LoginEvent
    data class Error(val errorMessage: UiText) : LoginEvent
}
```

And in your ViewModel:

```
private val eventsChannel = Channel<LoginEvent>()
val events = eventsChannel.receiveAsFlow()
```

For more info, Philipp talks about the problem in this video: https://youtu.be/njchj9d_Lf8?si=0to4KwRHzEIECID8